cupy/__init__.py

```diff
@@ -4,9 +4,19 @@
 import numpy
 import six
 
+try:
+    from cupy import core
+except ImportError:
+    # core is a c-extention module.
+    # When a user cannot import core, it represents that CuPy is not correctly
+    # built.
+    msg = ('CuPy is not correctly installed. Please check your environment, '
+           'uninstall Chainer and reinstall it with `pip install chainer '
+           '--no-cache-dir -vvvv`.')
+    raise RuntimeError(msg)
```

Comment:
Yes.
I suggest using `six.reraise`.
https://pythonhosted.org/six/#six.reraise

We want the original stack-trace when we develop CuPy.